### ISSUE TITLE
Feature: Add fades to credit screens

### ISSUE THREAD

**Issue by Richard-L** [rr-/Tomb1Main#476]
Now that thanks to #474 we have begun improving the game ending experience, and since we have also brought over the PS1 fading feature via #363, would it mean much trouble to add fades to the credit screens?

The PC versions of both TR1 and TR2 do not fade these. They're just harsh swaps like we have in TR1M too right now.

However on PS1:
TR1: https://youtu.be/ECVAdDHDssk?t=26562 (fade-in only)
TR2: https://youtu.be/93ceS2gbx68?t=25103 (fade-in + fade-out)

TR3 behaves like TR2, and @walkawayy had brought over my quick capture already:

https://user-images.githubusercontent.com/81546780/157767055-fcc498ad-0fb9-45d4-b378-a7b8184d7e36.mp4

^obviously we wouldn't have the loading bar, and I also don't think we need to half-fade the backdrop for the stats screen.

**Comment by walkawayy** [comment]
This is the last issue left before the 2.7 release I think. I still don't fully understand the fade effects, so I'm gonna defer to @rr- on this one. This change would likely have to go in `gameflow.c` for the `case GFS_DISPLAY_PICTURE:` case which means fades would happen for every picture. That might be OK but just a heads up.

**Comment by rr-** [comment]
Fade effects work by adding two overlays, one below all objects (inventory translucent background underneath the inventory items) and one above all objects (black overlay above game stats etc.), and only controlling its transparency. Additionally there is a notion of _target transparency_ and _current transparency_. Animating the fade updates the current transparency to "move" towards the target transparency a bit each frame. 

For example, in control_pause.c, we have:
```
    Output_FadeToSemiBlack(true);
    int32_t select = Control_Pause_Loop();
    Output_FadeToTransparent(true);
```

1. `Output_FadeToSemiBlack` sets the target fade value of the back overlay to 50% (its current transparency stays at 0%), or in case the fade effects are disabled, it immediately updates the _current_ transparency to 50%. 

2. Then we have the pause screen control loop. The control loop ultimately does `Output_DumpScreen()`. This in turns call `Output_FadeAnimate`, which updates the current transparency of all backdrops. Eventually over the course of running this loop, the background overlay transparency becomes 50%. This is our fade effect.

3. The user decides to leave the loop. We execute `Output_FadeToTransparent` which updates the target background transparency back to 0%, while the current transparency is still at 50%. We exit the function, most likely returning to regular game loop.

4. The regular game loop continues to call `Output_DumpScreen` which eventually updates the background current transparency to 0%.

This achieves the effect of fade animating while the menu is still opening, and animating after it gets closed "asynchronously" letting the player move around. It's a bit different with stats.c, which fade to black and _wait_ till the animation is finished. How is this accomplished?

```
    // finish fading
    while (Output_FadeIsAnimating()) {
        Output_InitialisePolyList();
        Draw_DrawScene(false);
        Output_DumpScreen();
    }
```

Basically we keep drawing the output scene until the fades stop animating, e.g. all current overlay transparency levels reach their target values.

---

LMK if this makes sense and is enough to get you going

**Comment by walkawayy** [comment]
Makes sense but does this all also apply to images? Instead of `Draw_DrawScene` in these fade loops do I need to `Output_CopyPictureToScreen` or something?

**Comment by rr-** [comment]
1. No, it's general and hopefully complete knowledge
2. Yes

In case the fade overlays do not play well with picture surfaces, please create a WIP PR and I'll help you get it to work.
